pkg/subscriptions/subscriptions.go

```diff
@@ -39,9 +38,18 @@ type subscriptionData struct {
 func (s subscriptionData) saveTo(dir string) error {
 	path := filepath.Join(dir, s.name)
 	if err := os.MkdirAll(filepath.Dir(path), s.dirMode); err != nil {
-		return err
+		return fmt.Errorf("create subscription directory: %w", err)
 	}
-	return os.WriteFile(path, s.data, s.mode)
+	if err := os.Chmod(dir, s.dirMode); err != nil {
```

Comment:
Is this necessary, since the MkdirAll already use the s.mode?

func [MkdirAll](https://cs.opensource.google/go/go/+/go1.20.3:src/os/path.go;l=18) [¶](https://pkg.go.dev/os#MkdirAll)
```
func MkdirAll(path [string](https://pkg.go.dev/builtin#string), perm [FileMode](https://pkg.go.dev/os#FileMode)) [error](https://pkg.go.dev/builtin#error)

MkdirAll creates a directory named path, along with any necessary parents, and returns nil, or else returns an error. The permission bits perm (before umask) are used for all directories that MkdirAll creates. If path is already a directory, MkdirAll does nothing and returns nil.
```

Although I am not sure what (before umask) means?

Reply:
it means the specified mask is the one you have before umask is applied.  We need the explicit Chmod as @Luap99 pointed out

Reply:
This is exactly why I was suggesting unit tests — not really because the _code_ is so uniquely complex that it needs tests that urgently, but because people do.